File: benchmarks/math/collatz/collatz.lua
-- Lua Benchmark: Collatz Sequence Max Length
function collatz(n)
    local len = 0
    while n > 1 do
        if n % 2 == 0 then
            n = math.floor(n / 2)
        else
            n = 3 * n + 1
        end
        len = len + 1
    end
    return len
end

local max_len = 0
for i = 1, 100000 do
    local clen = collatz(i)
    if clen > max_len then
        max_len = clen
    end
end

print("Max Collatz length up to 100000 = " .. max_len)

File: benchmarks/math/collatz/collatz.js
// Node.js Benchmark: Collatz Sequence Max Length
function collatz(n) {
  let len = 0;
  while (n > 1) {
    if (n % 2 === 0) {
      n = Math.floor(n / 2);
    } else {
      n = 3 * n + 1;
    }
    len++;
  }
  return len;
}

let maxLen = 0;
for (let i = 1; i <= 100000; i++) {
  const clen = collatz(i);
  if (clen > maxLen) {
    maxLen = clen;
  }
}

console.log(`Max Collatz length up to 100000 = ${maxLen}`);
